contracts/StablecoinConverter.sol

```diff
@@ -249,18 +257,23 @@ contract StablecoinConverter is EpochTokenLocker {
         }
     }
 
-    function getExecutedBuyAmount(
-        uint128 executedSellAmount,
+    function getExecutedSellAmount(
+        uint128 executedBuyAmount,
         uint128 buyTokenPrice,
         uint128 sellTokenPrice
     ) internal view returns (uint128) {
-        uint128 buyAmount = uint128(executedSellAmount.mul(sellTokenPrice) / buyTokenPrice);
-        // executedBuyAmount = buyAmount * (1 - (1/feeDenominator))
-        //                   = buyAmount - buyAmount/feeDenominator (*)
-        //                   = (buyAmount * feeDenominator)/ feeDenominator - buyAmount/feeDenominator
-        //                   = (buyAmount * feeDenominator - buyAmount) / feeDenominator
-        //                   = (buyAmount * (feeDenominator - 1)/feeDenominator
-        return uint128(buyAmount.mul(feeDenominator - 1) / feeDenominator);
+        // executedSellAmount = sellAmount * (1 - (1/feeDenominator))
+        //                   = sellAmount - sellAmount/feeDenominator (*)
+        //                   = (sellAmount * feeDenominator)/ feeDenominator - sellAmount/feeDenominator
+        //                   = (sellAmount * feeDenominator - sellAmount) / feeDenominator
+        //                   = (sellAmount * (feeDenominator - 1)/feeDenominator
+        //                   = (executedBuyAmount * buyTokenPrice / sellTokenPrice) * (feeDenominator - 1) / feeDenominator
+        //                   in order to minimize rounding errors, the order is switched
+        //                   = (executedBuyAmount * buyTokenPrice / feeDenominator) * (feeDenominator - 1) / sellTokenPrice
+        uint256 sellAmount = (uint256(executedBuyAmount).mul(buyTokenPrice) / (feeDenominator - 1))
+            .mul(feeDenominator) / sellTokenPrice;
+        require(sellAmount < MAX_UINT128, "downcasting from u256 to u128 would change the value");
```

Comment:
nit: How about simply saying "sellAmount too large" ?

Reply:
Also, is MAX_UINT128 = 2**128 - 1, then this requirement should include equality! (`<=`)